File: 2013/Lunch & Learn/IntroToOOP/C#/Example2/Cat.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace IntroToOOP.Example2
{
    public class Cat
    {            
        /* Constructor */
        public Cat()
        {
            /* State */
            this.Name = "Sue";
            this.Color = "Black";
        }

        /* Consctructor with parameters (different signature) */
        public Cat(string name, string color)
        {
            Name = name;
            Color = color;
        }        

        /* Public properties to allow the return and modification of the private members
         * Allow the State to be changed from outside the class*/
        public string Name { get; set;}
               
        public string Color { get; set;}           

        public void wagsTail()
        {
            Console.WriteLine("{0} wags tail...", Name);
        }

        public virtual void Speaks()
        {
            Console.WriteLine("{0} purrs...", Name);
        }

    }

    public class OohCat : Cat
    {         
        public override void Speaks()
        {
            Console.WriteLine("{0} raises hand to mouth slowly and says OOOOOHHH", Name);
        }
    }
}

File: 2013/Lunch & Learn/IntroToOOP/VB.NET/IntroToOOP/Example2/Cat.vb

Namespace IntroToOOP.Example2
    Public Class Cat

        Public Sub New()
            Me.Name = "Sue"
            Me.Color = "Black"
        End Sub

        Public Sub New(name As String, color As String)
            Me.Name = name
            Me.Color = color
        End Sub

        Public Name As String
        Public Color As String

        Public Sub wagsTail()
            Console.WriteLine("{0} wags tail...", Name)
        End Sub

        Public Overridable Sub Speaks()
            Console.WriteLine("{0} purrs...", Name)
        End Sub

    End Class

    Public Class OohCat
        Inherits Cat

        Public Overrides Sub Speaks()
            Console.WriteLine("{0} raises hand to mouth slowly and says OOOOOOOOHHHHHH!", Name)
        End Sub

    End Class

End Namespace
